File: Python/String/E-28. Find the Index of the First Occurrence in a String.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if len(needle) > len(haystack):
            return -1

        n, m = len(haystack), len(needle)

        # sadbutsad, sad => n = 8, m = 3
        for i in range(n):
            if i + m > n:
                return -1
            is_substr = True
            for j in range(m):
                if haystack[i + j] != needle[j]:
                    is_substr = False

            if is_substr:
                return i

        return -1

File: Javascript/String/E-28. Find the Index of the First Occurrence in a String.ts
function strStr(haystack: string, needle: string): number {
  if (needle.length > haystack.length) return -1;
  for (let i = 0; i < haystack.length; i++) {
    let isSubstr = true;
    for (let j = 0; j < needle.length; j++) {
      if (haystack[i + j] !== needle[j]) isSubstr = false;
    }
    if (isSubstr) return i;
  }
  return -1;
}
